pkg/metrics/statsd/statsd.go

```diff
@@ -103,13 +108,42 @@ func (s *statsd) run() {
 	}
 }
 
+func (s *statsd) getConnectionDetails() (protocol string, connectionstring string, err error) {
+	// allow the socket connection to be overriden via ENV Variable
+	o, isset := os.LookupEnv(STATSD_SOCKET_ENV)
+	if !isset { //original behaviour
+		protocol = "unix"
+		connectionstring = "/var/etw/mdm_statsd.socket"
+		if s.env.IsLocalDevelopmentMode() {
+			connectionstring = "mdm_statsd.socket"
+		}
+
+		return
+	}
+
+	// Assume protocol:connectionstring format
+	a := strings.SplitN(o, ":", 2)
+	if len(a) != 2 {
+		err = errors.New("malformated ENV Variable ARO_STATSD_SOCKET. Expecting udp:host:port or unix:path-to-socket format")
+
+		return
+	}
+	protocol = strings.ToLower(a[0])
+	connectionstring = a[1]
+
+	return
```

Comment:
This could probably its own function so we can test it separately. 

Also, why do you do `a := strings.SplitN(o, ":", 2)` if you check afterwards again if its 2 ? Something like udp:a:a:a:a:a pass your checks but would not be valid ?

Reply:
`Malformated` should be `malformatted` I think. I personally would use `malformed` but I don't really know if there is an actual difference of meaning, so not asking to change :)

Reply:
I need the part before the first separator as protocol and then rest of the string as connectionstring to pass to the net.Dial call. I let the evaluation of the actual values of protocol  and connectionstring to the Dial call, I don't want to duplicate that logic and probably get it wrong. I could use two environment variables and save the parsing here, but I opted to mimic how socat is handling these parameters.

Now, SplitN(o,":",2) could deliver 0,1 or 2 items in the slice it returns, so I need to check if I have enough items to go forward. I'll could change it to <2, if that is clearer.

So yes, your example would pass my test, but fail in the dial call.

Reply:
I understand what the logic is , but it's not valid, since it accepts both wrong protocols / malformed urls . 

BTW I don't know what the policy is around here, but I think it would be clearer if you used `return value1, value2,value3` instead of just `return`

Reply:
what do you think about that ?
```go
func validateEnv(env string) error {
	if strings.HasPrefix(env, "udp:") && len(strings.Split(env, ":")) == 3 {
		return nil
	}
	if strings.HasPrefix(env, "unix:") && len(strings.Split(env, ":")) >= 2 {
		return nil
	}

	return fmt.Errorf("invalid env value")
}
```

Reply:
Added additional input validation, split the logic up a bit and updated the unit tests accordingly